manager/integration/framework/node_exec/node_exec.py

```diff
@@ -0,0 +1,197 @@
+            stdout=True,
+            tty=False
+        )
+        return res
+
+    def launch_pod(self, node_name):
+        if node_name in self.node_exec_pod:
+            return self.node_exec_pod[node_name]
+        else:
+            pod_manifest = {
+                'apiVersion': 'v1',
+                'kind': 'Pod',
+                'metadata': {
+                    'name': node_name
+                },
+                'spec': {
+                    'affinity': {
+                        'nodeAffinity': {
+                            'requiredDuringSchedulingIgnoredDuringExecution': {
+                                'nodeSelectorTerms': [{
+                                    'matchExpressions': [{
+                                        'key': 'kubernetes.io/hostname',
+                                        'operator': 'In',
+                                        'values': [
+                                            node_name
+                                        ]
+                                    }]
+                                }]
+                            }
+                        }
+                    },
+                    'containers': [{
+                        'image': 'busybox:1.34.0',
+                        'imagePullPolicy': 'IfNotPresent',
+                        'securityContext': {
+                            'privileged': True
+                        },
+                        'name': 'node-exec',
+                        "args": [
+                            "tail", "-f", "/dev/null"
+                        ],
+                        "volumeMounts": [{
+                            'name': 'rootfs',
+                            'mountPath': '/rootfs',
+                            'readOnly': True
+                        }],
+                    }],
+                    'volumes': [{
+                        'name': 'rootfs',
+                        'hostPath': {
+                            'path': '/'
+                        }
+                    }]
+                }
+            }
+            pod = self.core_api.create_namespaced_pod(
+                body=pod_manifest,
+                namespace=self.namespace
+            )
+            for i in range(DEFAULT_POD_TIMEOUT):
+                pod = self.core_api.read_namespaced_pod(
+                        name=node_name,
+                        namespace=self.namespace
+                      )
+                if pod is not None and pod.status.phase == 'Running':
+                    break
+                time.sleep(DEFAULT_POD_INTERVAL)
+            self.node_exec_pod[node_name] = pod
+            return pod
+    
+    def power_off_node(self, node_name):
+        self._instance.power_off_node_instance(node_name)
+    
+    def power_on_node(self, node_name):
+        self._instance.power_on_node_instance(node_name)
+
+    def restart_kubelet(self, node_name, interval_time):
+        if node_name == "":
```

Comment:
Will we need to restart all worker nodes' kubelet all at once?

Reply:
I think you are right. The scenario that restarts kubelet of all nodes should not exist. I will raise an exception if the parameter `node_name` is empty. Thank you.